File: backend/calculations.py
import math


def calculate_job(
    order_quantity: int,
    ups: int,
    sheet_length: float,
    sheet_width: float,
    gsm: int,
) -> dict:
    base_sheets = math.ceil(order_quantity / ups)

    if base_sheets < 5000:
        wastage_pct = 10.0
    elif base_sheets <= 7500:
        wastage_pct = 7.0
    else:
        wastage_pct = 5.0

    final_sheets = math.ceil(base_sheets * (1 + wastage_pct / 100))
    total_kg = round(
        ((sheet_length * sheet_width * gsm) / 20000 / 500) * final_sheets, 2
    )

    return {
        "base_sheets": base_sheets,
        "wastage_percentage": wastage_pct,
        "final_sheets": final_sheets,
        "total_kg": total_kg,
    }

File: frontend/src/lib/calculations.ts
import { CalcResult } from '@/types';

export function calculateJob(
  orderQty: number,
  ups: number,
  sheetL: number,
  sheetW: number,
  gsm: number,
): CalcResult | null {
  if (!orderQty || !ups || !sheetL || !sheetW || !gsm) return null;
  if (orderQty <= 0 || ups <= 0 || sheetL <= 0 || sheetW <= 0 || gsm <= 0) return null;

  const baseSheets = Math.ceil(orderQty / ups);

  let wastagePct: number;
  if (baseSheets < 5000) wastagePct = 10;
  else if (baseSheets <= 7500) wastagePct = 7;
  else wastagePct = 5;

  const finalSheets = Math.ceil(baseSheets * (1 + wastagePct / 100));
  const totalKg = parseFloat(
    (((sheetL * sheetW * gsm) / 20000 / 500) * finalSheets).toFixed(2),
  );

  return { base_sheets: baseSheets, wastage_percentage: wastagePct, final_sheets: finalSheets, total_kg: totalKg };
}
